config/config-mp/src/main/java/io/helidon/config/mp/MpConfigImpl.java

```diff
@@ -0,0 +1,392 @@
+     *
+     * This method is from a future version of MP Config specification and may changed before it
+     * is released. It is nevertheless needed to process annotations with default values.
+     *
+     * @param <T> the conversion type
+     * @param forType the type to be produced by the converter
+     * @return an {@link java.util.Optional} containing the converter, or empty if no converter is available for the specified type
+     */
+    @SuppressWarnings("unchecked")
+    public <T> Optional<Converter<T>> getConverter(Class<T> forType) {
+        return converters.entrySet()
+                .stream()
+                .filter(it -> forType.isAssignableFrom(it.getKey()))
+                .findFirst()
+                .map(Map.Entry::getValue)
+                .map(it -> (Converter<T>) it)
+                .or(() -> findImplicit(forType));
+    }
+
+    /**
+     * Convert a String to a specific type.
+     * This is a helper method to allow for processing of default values that cannot be typed (e.g. in annotations).
+     *
+     * @param propertyName name of the property, used for error messages
+     * @param type  type of the property
+     * @param value String value (may be null)
+     * @param <T>   type
+     * @return instance of the correct type, may return null in case null was provided and converter did not do this
+     * @throws IllegalArgumentException in case the String provided cannot be converted to the type expected
+     */
+    private <T> T convert(String propertyName, Class<T> type, String value) {
+        try {
+            return findConverter(type)
+                    .convert(value);
+        } catch (Exception e) {
+            throw new IllegalArgumentException("Failed to convert property \""
+                                                       + propertyName
+                                                       + "\" from its value \""
+                                                       + value
+                                                       + "\" to "
+                                                       + type.getName(),
+                                               e);
+        }
+    }
+
+    private Optional<String> getStringValue(String propertyName) {
+        for (ConfigSource source : sources) {
+            String value = source.getValue(propertyName);
+
+            if (null == value) {
+                // not in this one
+                continue;
+            }
+
+            LOGGER.finest("Found property " + propertyName + " in source " + source.getName());
+            return Optional.of(resolveReferences(propertyName, value));
+        }
+
+        return Optional.empty();
+    }
+
+    private Optional<String> applyFilters(String propertyName, String stringValue) {
+        String result = stringValue;
+
+        for (MpConfigFilter filter : filters) {
+            result = filter.apply(propertyName, stringValue);
```

Comment:
`filter.apply(propertyName, result)` or is it last filter wins?

Reply:
correct!